cmd/policy.go

```diff
@@ -88,7 +87,15 @@ func (sys *PolicySys) IsAllowed(args policy.Args) bool {
 	if p, found := sys.bucketPolicyMap[args.BucketName]; found {
 		return p.IsAllowed(args)
 	}
-
+	objAPI := newObjectLayerFn()
+	if objAPI != nil {
+		config, err := objAPI.GetBucketPolicy(context.Background(), args.BucketName)
+		if err == nil {
+			// Update the cache with new policy.
+			sys.bucketPolicyMap[args.BucketName] = *config
```

Comment:
Why is the policy being cached, I thought the idea is to go to the backend to check the policy when we need to check. We might not have to check the policy for all the calls anyway

Reply:
Oh yes it is better to hit the backend..